src/Microsoft.Extensions.FileProviders.Physical/Watcher/PollingFileWatcher.cs

```diff
@@ -0,0 +1,243 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+
+namespace Microsoft.Extensions.FileProviders.Physical.Watcher
+{
+    internal class PollingFileWatcher : IFileSystemWatcher
```

Comment:
Can you do some profiling on this before committing it? Pick an application that has npm modules installed. Storing all files in a dictionary screams badness.